plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -140,11 +140,17 @@ private static void sendPostRequest(String parametersString) {
       connection.setDoOutput(true);
       connection.setRequestMethod("POST");
       connection.setRequestProperty("Content-Length", Integer.toString(parametersString.length()));
+      connection.setReadTimeout(3000);  // milliseconds
 
       try (Writer writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8")) {
         writer.write(parametersString);
         writer.flush();
       }
+
+      int responseCode = connection.getResponseCode();
+      if (responseCode != 200) {
+        logger.log(Level.WARNING, "Response code not 200: " + responseCode);
```

Comment:
I'm not sure we really need to log this, but if we do specify where this is from. E.g.

Analytics response code: